### ISSUE TITLE
lint: `unreachable_from_main` has false positives with typedefs

### ISSUE THREAD

**Issue by arthurbcd** [dart-lang/linter#4911]
**Describe the issue**

Turn on the lint:
```yaml
linter:
  rules:
    unreachable_from_main: true
```

Define a typedef for a widget and use it inside `main`:
```dart
void main() {
  runApp(const MyApp());
}

// LINT: unreachable_from_main
typedef MyApp = MaterialApp; <- triggers lint
```

**Expected results**

No lint warning, as Im using the typedef inside main.

The lint says it's not being used and therefore it's a false positive.

**Comment by bwilkerson** [comment]
@srawlins

**Comment by srawlins** [comment]
Good catch!

**Comment by srawlins** [comment]
Fixed by https://dart-review.googlesource.com/c/sdk/+/357182
